egs2/iam/ocr1/local/data.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+# Set bash to 'debug' mode, it will exit on :
+# -e 'error', -u 'undefined variable', -o ... 'error in pipeline', -x 'print commands',
+set -e
+set -u
+set -o pipefail
+
+log() {
+    local fname=${BASH_SOURCE[1]##*/}
+    echo -e "$(date '+%Y-%m-%dT%H:%M:%S') (${fname}:${BASH_LINENO[0]}:${FUNCNAME[1]}) $*"
+}
+
+SECONDS=0
+
+. ./db.sh
+. ./path.sh
+. ./cmd.sh
+
+stage=1
+stop_stage=2
+
+# Fill in username/password from account on https://fki.tic.heia-fr.ch/register
+iam_username=""
+iam_password=""
+
+# Set parameters for the feature dimensions used during image extraction,
+# see data_prep.py for details
+feature_dim=100
+downsampling_factor=0.5
+
+data_dir=data/
+
+if [ ${stage} -le 1 ] && [ ${stop_stage} -ge 1 ]; then
+    log "Stage 1.1: Downloading the IAM Handwriting dataset with username ${iam_username} and password ${iam_password}"
+    mkdir -p ${IAM}
```

Comment:
Can you follow some other recipes (e.g., commonvoice) to check if the `IAM` value has been set?

Reply:
not sure exactly what you mean by this - do you just mean if the $IAM variable is set after running ./db.sh in another recipe? If so, I checked by adding an echo $IAM in the commonvoice data.sh and confirmed that it was set.

Reply:
@ftshijt is asking you to follow https://github.com/espnet/espnet/blob/master/egs2/commonvoice/asr1/local/data.sh#L27-L31
and add the following lines
```
mkdir -p ${IAM}
if [ -z "${IAM}" ]; then
    log "Fill the value of 'IAM' of db.sh"
    exit 1
fi
```

Reply:
Ahh ok understood and updated!